engine/src/main/scala/cromwell/engine/workflow/lifecycle/EngineWorkflowInitializationActor.scala

```diff
@@ -0,0 +1,118 @@
+  case object AbortInitialization extends EngineWorkflowInitializationActorCommand
+
+  /**
+    * Responses
+    */
+  sealed trait EngineWorkflowInitializationActorResponse
+  case object WorkflowInitializationSucceeded extends EngineWorkflowInitializationActorResponse
+  case object WorkflowInitializationAborted extends EngineWorkflowInitializationActorResponse
+  final case class WorkflowInitializationFailed(reasons: Map[ActorRef, Throwable]) extends EngineWorkflowInitializationActorResponse
+
+  def props(workflowTag: String, backendAssignments: Map[Call, String]): Props = Props(new EngineWorkflowInitializationActor(workflowTag, backendAssignments))
+
+}
+
+case class EngineWorkflowInitializationActor(workflowTag: String, backendsToInitialize: Map[Call, String]) extends LoggingFSM[EngineWorkflowInitializationActorState, EngineWorkflowInitializationActorData] {
+
+  startWith(Unstarted, EngineWorkflowInitializationActorData(Map.empty, List.empty, Map.empty))
+
+  when(Unstarted) {
+    case Event(StartInitialization, _) =>
+      val backendInitializationActors = backendWorkflowInitializationActors // Create each InitializationActor with the list of calls assigned to it.
+      backendInitializationActors.keys foreach { _ ! Initialize }
+      goto(InitializingBackends) using EngineWorkflowInitializationActorData(backendInitializationActors, List.empty, Map.empty)
+    case Event(AbortInitialization, _) =>
+      context.parent ! WorkflowInitializationAborted
+      goto(InitializationsAborted)
+  }
+
+  when(InitializingBackends) {
+    case Event(InitializationSuccess, stateData) =>
+      val newData = stateData.copy(successes = stateData.successes ++ List(sender))
+      checkForDoneAndTransition(newData)
+    case Event(InitializationFailed(reason), stateData) =>
+      val newData = stateData.copy(failures = stateData.failures ++ Map(sender -> reason))
+      checkForDoneAndTransition(newData)
+    case Event(AbortInitialization, _) => ??? // TODO: Handle this
+  }
+
+  private def checkForDoneAndTransition(newData: EngineWorkflowInitializationActorData): State = {
+    if (checkForDone(newData)) {
+      if (newData.failures.isEmpty){
+        context.parent ! WorkflowInitializationSucceeded
+        goto(InitializationSucceededState)
+      }
+      else {
+        context.parent ! WorkflowInitializationFailed(newData.failures)
+        goto(InitializationFailedState) using newData
+      }
+    }
+    else {
+      stay using newData
+    }
+  }
+
+  private def checkForDone(stateData: EngineWorkflowInitializationActorData) = {
+    // TODO: Check these are actually the same?
+    val allActorResponses = stateData.successes ++ stateData.failures.keys
+    allActorResponses.size == backendsToInitialize.size
+  }
+
+  whenUnhandled {
+    case unhandledMessage =>
+      log.info(s"Backends initializer for $workflowTag received an unhandled message: $unhandledMessage")
```

Comment:
We shouldn't be seeing any unhandled messages if I'm not mistaken... so maybe `log.warn`?